presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java

```diff
@@ -1296,45 +1299,143 @@ public void testSequence()
+                new ArrayType(TIMESTAMP),
+                ImmutableList.of(sqlTimestamp("2016-04-16 01:00:10"), sqlTimestamp("2016-04-16 01:03:10"), sqlTimestamp("2016-04-16 01:06:10")));
+        assertFunction(
+                "SEQUENCE(timestamp '2016-04-16 01:10:10', timestamp '2016-04-16 01:03:00', interval '-3' minute)",
+                new ArrayType(TIMESTAMP),
+                ImmutableList.of(sqlTimestamp("2016-04-16 01:10:10"), sqlTimestamp("2016-04-16 01:07:10"), sqlTimestamp("2016-04-16 01:04:10")));
+
+        assertFunction(
+                "SEQUENCE(timestamp '2016-04-16 01:00:10', timestamp '2016-04-16 01:01:00', interval '20' second)",
+                new ArrayType(TIMESTAMP),
+                ImmutableList.of(sqlTimestamp("2016-04-16 01:00:10"), sqlTimestamp("2016-04-16 01:00:30"), sqlTimestamp("2016-04-16 01:00:50")));
+        assertFunction(
+                "SEQUENCE(timestamp '2016-04-16 01:01:10', timestamp '2016-04-16 01:00:20', interval '-20' second)",
+                new ArrayType(TIMESTAMP),
+                ImmutableList.of(sqlTimestamp("2016-04-16 01:01:10"), sqlTimestamp("2016-04-16 01:00:50"), sqlTimestamp("2016-04-16 01:00:30")));
 
-        assertFunction("SEQUENCE(date '2016-04-12', date '2016-04-16', interval '2' day)", new ArrayType(TIMESTAMP),
-                ImmutableList.of(sqlTimestamp("2016-04-12"), sqlTimestamp("2016-04-14"), sqlTimestamp("2016-04-16")));
-        assertFunction("SEQUENCE(date '2016-04-16', date '2016-04-12', interval '-2' day)", new ArrayType(TIMESTAMP),
-                ImmutableList.of(sqlTimestamp("2016-04-16"), sqlTimestamp("2016-04-14"), sqlTimestamp("2016-04-12")));
+        assertFunction(
+                "SEQUENCE(timestamp '2016-04-16 01:00:10', timestamp '2016-04-18 01:01:00', interval '19' hour)",
+                new ArrayType(TIMESTAMP),
+                ImmutableList.of(sqlTimestamp("2016-04-16 01:00:10"), sqlTimestamp("2016-04-16 20:00:10"), sqlTimestamp("2016-04-17 15:00:10")));
+        assertFunction(
+                "SEQUENCE(timestamp '2016-04-16 01:00:10', timestamp '2016-04-14 01:00:20', interval '-19' hour)",
+                new ArrayType(TIMESTAMP),
+                ImmutableList.of(sqlTimestamp("2016-04-16 01:00:10"), sqlTimestamp("2016-04-15 06:00:10"), sqlTimestamp("2016-04-14 11:00:10")));
 
         // failure modes
-        assertInvalidFunction("SEQUENCE(date '2016-04-12', date '2016-04-14', interval '-1' day)", INVALID_FUNCTION_ARGUMENT);
-        assertInvalidFunction("SEQUENCE(date '2016-04-14', date '2016-04-12', interval '1' day)", INVALID_FUNCTION_ARGUMENT);
-        assertInvalidFunction("SEQUENCE(date '2000-04-14', date '2030-04-12', interval '1' day)", INVALID_FUNCTION_ARGUMENT);
+        assertInvalidFunction(
+                "SEQUENCE(date '2016-04-12', date '2016-04-14', interval '-1' day)",
+                INVALID_FUNCTION_ARGUMENT);
```

Comment:
Can you change all of these to have the expected error message rather than `INVALID_FUNCTION_ARGUMENT`?